### PR TITLE
fix: ts esm type issues

### LINKED ISSUE #3817: Bug: type issue in ts@4.8+ with esm
### Description

The package is shipping esm files, but package.json is missing "type": "module"

So typescript will assume users are importing a commonjs module, so it will report issues with:

```ts
import memaid from "mitt";

memaid.someAPI(...args)
```

Saying the expression is not callable.

That's because a cjs module will have it's module.exports on default key, so:

```ts
import memaid from "mitt";

memaid.default.someAPI(...args)
```

is ok to typescript, but it's not acually working (because we do have esm modules using export default) and the default key does not exisit.

### Steps to reproduce

Create project with `"type": "module"` and install mermaid. then ensure ts.config.json has `target: "ESxxx"`, you will get issues mention above. (Does not need to use `moduleResolution: NodeNext`)

### Screenshots

_No response_

### Code Sample

_No response_

### Setup

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Mister-Hope**
## :bookmark_tabs: Summary

Brief description about the content of your PR.

Resolves #3817

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :bookmark: targeted `develop` branch

---

Warnning: I am not pretty sure if it will have effects on edge case, for a package adding esm modules, raising a major version **is expected**, however mermaid fails to do this.

### CODE REVIEW

**Comment by sidharthv96** [comment]
This was specifically reverted as a fix in https://github.com/mermaid-js/mermaid/pull/3767.

@aloisklink

**Review by aloisklink — CHANGES_REQUESTED** [review]
We added `"type": "module"` in Mermaid v9.2.0, but this then broke Mermaid when using TypeScript for CommonJS projects See https://github.com/mermaid-js/mermaid/issues/3747 for the bug report and https://github.com/mermaid-js/mermaid/pull/3767 for the PR undoing it.

We could delete the `"type": ...` completely, which means it should default to "type": "commonjs" for most projects, but it might work for some bundlers, see https://github.com/mermaid-js/mermaid/issues/3799#issuecomment-1315703448.

Because of that, we won't merge this unless we also drop CommonJS support, which would be a breaking change, and may also need to wait until Docusaurus switches to ESM.

Can you try using `"compilerOptions": {"module": "node16"}` in your tsconfig.json file? It should let you load the ESM version when doing `import mermaid from "mermaid". See https://www.typescriptlang.org/docs/handbook/esm-node.html for more information.

**Pull request closed without merging**
